views/js/runner/proxy/cache/proxy.js

```diff
@@ -439,35 +475,42 @@ define([
         /**
          * Sends the test variables
          * @param {Object} variables
+         * @param {Boolean} deferred whether action can be scheduled (put into queue) to be sent in a bunch of actions later.
          * @returns {Promise} - Returns a promise. The result of the request will be provided on resolve.
          *                      Any error will be provided if rejected.
          * @fires sendVariables
          */
-        sendVariables: function sendVariables(variables) {
+        sendVariables: function sendVariables(variables, deferred) {
             var action = 'storeTraceData';
             var actionParams = {
                 traceData: JSON.stringify(variables)
             };
 
+            deferred = !!deferred;
```

Comment:
so those action are `deferered` by default ?

Reply:
`!!` is used to cast value to boolean type. If the second variable is not given then `deferred` is `false`:
```
!!undefined //false
```

Reply:
please remove those casts, they are useless, the value is  only read at https://github.com/oat-sa/extension-tao-testqti/pull/1085/files#diff-acdd119a84a964e59f062fae31283a2fR284 and is already casted/coerced there.